### PR TITLE
[BT-2] Implement core project

### PR DESCRIPTION
**Pull request by M-Vitalii**
# Description
Implemented Core project

### What's added:
- Contracts
- Mapping
- Validation
- Interfaces for service and repository layer

### CODE REVIEW

File: BillTrack.Core/Interfaces/Services/IDepartmentService.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+using BillTrack.Core.Contracts.Department;
+
+namespace BillTrack.Core.Interfaces.Services;
+
+public interface IDepartmentService : IGenericService<DepartmentResponse, DepartmentRequest>
```
Comment (PIoneer2) [inline BillTrack.Core/Interfaces/Services/IDepartmentService.cs]: Typed services are not required. Services are for business logic. They include repo operations and more. They can operate with any incoming or outgoing objects.
This project can have just a few non-typed services. Each of them will have operations with different entities.

File: BillTrack.Core/Interfaces/Services/IGenericService.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+namespace BillTrack.Core.Interfaces.Services;
+
+public interface IGenericService<TResponse, in TRequest>
```
Comment (PIoneer2) [inline BillTrack.Core/Interfaces/Services/IGenericService.cs]: There is IGenericRepository to handle with Entities. Also, services are not typed. Then such a generic service is not required either.

**Review by PIoneer2 — CHANGES_REQUESTED** [review]
Please, remove IGenericService and the typed services.
For now, create 3 services. 1 service for each of our applications.

File: BillTrack.Core/Interfaces/Services/IDepartmentService.cs (hunk 1 above)
Comment (M-Vitalii) [reply BillTrack.Core/Interfaces/Services/IDepartmentService.cs]: fixed

File: BillTrack.Core/Interfaces/Services/IGenericService.cs (hunk 2 above)
Comment (M-Vitalii) [reply BillTrack.Core/Interfaces/Services/IGenericService.cs]: fixed

**Comment by M-Vitalii** [comment]
The validation will be transferred to the Web Api project when the task “Implement the Web Api project” starts.

**Review by PIoneer2 — APPROVED** [review]
implement bases for the Core project.
OK.

**Pull request merged**